doc/getting-started/generating-transactions.md

```diff
@@ -1,14 +1,105 @@
 # Generating Transactions
 
-To mention:
+## TransactionBuilder API
 
-1) TransactionBuilder interface
-2) Why fee calculation differs in Rust vs Haskell
-3) Change address calculation
+In order to simplify transaction creation, we provide a `TransactionBuilder` struct that manages witnesses, fee calculation, change addresses and such. Assume we have instantiated an instance under the variable `builder` for this explanation.
 
-Need to tell people to refer to the spec for:
+The minimum required for a valid transaction is to add inputs, outputs, time-to-live and either set the fee explicitly with `builder.set_fee(fee)`, or calculate it implicitly using `builder.add_change_if_needed(address)`.
+Optionally a transaction can also have certificates, withdrawals, and metadata added to it.
+Any change made to the builder can impact the size and thus the fee so the fee should be the last thing set.
+If implicitly setting the fee any extra ADA (`inputs + withdrawals - outputs - min fee`) is sent to the provided change address.
```

Comment:
This is missing `+ refund - deposit`

Probably we should mentioned refunds and deposits somewhere in this file with a link to learn more?